test/functional/autocmd/autocmd_spec.lua

```diff
@@ -34,6 +36,76 @@ describe('autocmds:', function()
     assert.same(expected, eval('g:foo'))
   end)
 
+  it(':close triggers WinClosed event', function()
+    command('let g:triggered = 0')
+    command('new')
+    command('autocmd WinClosed <buffer> :let g:triggered+=1')
+    eq(0, eval('g:triggered'))
+    command('close')
+    eq(1, eval('g:triggered'))
+  end)
+
+  it(':bdelete triggers WinClosed event', function()
+    command('let g:triggered = 0')
+    command('autocmd WinClosed <buffer> :let g:triggered+=1')
+    local first_buffer = eval('bufnr(\'%\')')
```

Comment:
can avoid escaping by using double quotes (interchangeable with single-quotes, in Lua).

    local first_buffer = eval("bufnr('%')")